scalameta/parsers/js/src/main/scala/scala/meta/parsers/JSFacade.scala

```diff
@@ -56,22 +66,41 @@ object JSFacade {
       case _ => js.Dynamic.literal()
     }
 
-    mergeJSObjects(base, value, syntax)
+    mergeJSObjects(base, fields, value, syntax)
   }
 
-  private[this] def parse[A <: Tree: Parse](code: String): js.Dictionary[Any] =
-    code.parse[A] match {
+  private[this] type Settings = js.UndefOr[js.Dictionary[String]]
+  private[this] val defaultSettings = None.orUndefined
+
+  private[this] def parse[A <: Tree: Parse](
+    code: String,
+    settings: Settings
+  ): js.Dictionary[Any] = {
+    val dialect = (for {
+      settings <- settings.toOption
+      dialectStr <- settings.get("dialect")
+      dialect <- Dialect.standards.get(dialectStr)
+    } yield dialect).getOrElse(dialects.Scala211)
```

Comment:
I wanted to keep it consistent with the current meta behavior. Is that going to change?